packages/sanity/src/core/form/inputs/files/ImageInput/ImageInput.tsx

```diff
@@ -303,15 +275,16 @@ function BaseImageInputComponent(props: BaseImageInputProps): JSX.Element {
     menuButtonElement?.focus()
   }, [menuButtonElement])
 
-  const renderPreview = useCallback(() => {
+  const renderPreview = useCallback<() => JSX.Element>(() => {
```

Comment:
I think it depends on how you feel about inference. It can be valuable IMO to declare what a type _should_ be, rather than relying on inference.

`renderPreview` is passed to `ImageInputAsset`, which expects a function returning a `JSX.Element`.

Without setting the type parameter for the `useCallback`, somebody may think they can [return `null` on this line](https://github.com/sanity-io/sanity/pull/7535/files/2eb20ff932e53d8012a7ab721672b30743611d6e#diff-096da0579fec92502d5b032482596c3aa04075874503674869fa01e7313b0358R280), only to cause a type error elsewhere in the code.

```diff
if (!value) {
   // okay
-  return <></>
   // not okay
+  return null
}
```

Happy to reconsider, though 🙂.